pkg/workerapi/workerserver/rpcserver.go

```diff
@@ -0,0 +1,84 @@
+package workerserver
+
+import (
+	"errors"
+	"net"
+	"sync"
+	"time"
+
+	v1 "github.com/iver-wharf/wharf-cmd/api/workerapi/v1"
+	"github.com/iver-wharf/wharf-cmd/pkg/resultstore"
+	"google.golang.org/grpc"
+)
+
+type rpcServer struct {
+	bindAddress string
+
+	grpcServer   *grpc.Server
+	workerServer *workerRPCServer
+	isRunning    bool
+	serveMutex   *sync.Mutex
+}
+
+// NewRPCServer creates a new server that can be started by calling Serve.
+func NewRPCServer(bindAddress string, store resultstore.Store) Server {
```

Comment:
nit: This whole package is for providing an RPC server, including the HTTP stuff. What you're probably referring to here is gRPC in particular.

Yes, NewGRPCServer doesn't look as nice, especially with gRPC canonical spelling of the lowercase `g`, but only calling it RPC makes the HTTP client and server stand out as non-RPC? :)

However, to restate, I think you should just merge the two and then call this function simply `func New()`.